src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -317,18 +317,20 @@ internal void Connect(ServerInfo serverInfo, SqlInternalConnectionTds connHandle
                 Debug.Assert(false, "SNI returned status != success, but no error thrown?");
             }
 
-#if !MANAGED_SNI
+
             if (integratedSecurity)
             {
+#if !MANAGED_SNI
                 LoadSSPILibrary();
+#endif
                 // now allocate proper length of buffer
                 _sniSpnBuffer = new byte[SNINativeMethodWrapper.SniMaxComposedSpnLength];
```

Comment:
```
private SNITCPHandle CreateTcpHandle(string fullServerName, long timerExpire, object callbackObject, bool parallel, ref byte[] spnBuffer)
{
...
}
```
In Managed SNI,  `CreateTcpHandle()` method can know if Integrated Authentication is being used by looking at if _sniSpnBuff is null or not. This is the only way to forward the hint to `CreateTcpHandle()` method now.